### ISSUE TITLE
List of bugs from latest build. 

### ISSUE THREAD

**Issue by matt-bel** [FlansMods/FlansMod#1048]
**Critical Errors**
1. Destroying a vehicle with players in it will cause the players inside the vehicle to freeze in some strange sort of limbo. 
2. Passenger guns for most vehicles generally do not work. This is not consistent, but the cooldown timer for hitting/using objects introduced in 1.12.2 seems to be the culprit. I plan to try to use flans mod in conjunction with something that removes the cooldown to see if that changes anything. In general, shooting/controlling vehicles is very inconsistent and bugged. 

**Graphical/Minor Errors**
1. ~MG42 placed machinegun does not properly render, the model is a bunch of stripes.~
2. Placed MG's do not play the animation for their ammo loading and ~take an entire stack of ammo at a time, instead of one clip at a time.~
3. Leopard Tank has a lot of issues - gun not firing AP shells, etc.
4. The newly added Russian self-propelled artillery references a shell type that doesn't exist. easy fix would be to just use HE ammo for it, but I can provide a texture for an arty shell below.
![tankshellhe](https://user-images.githubusercontent.com/13383263/57170764-28ba2a80-6ddd-11e9-8712-d4ac696bfdcf.png)
![tankshellhe2](https://user-images.githubusercontent.com/13383263/57170765-28ba2a80-6ddd-11e9-8ba0-79f06978167b.png)

**Miscellaneous Errors**
1. I personally did not witness this one, but server staff reported numerous server crashes related to:
`com.flansmod.common.driveables.mechas.EntityMecha.func_70071h(EntityMecha.java:581)`
Specifically, they were testing out zero titans and left a bunch around the map. In the end, the world they were in had to be reset, and certain players had their playerdata corrupted. 

**Versions (please complete the following information):**
 - Forge: forge-1.12.2-14.23.5.2768
 - Flan's Mod: Latest build off github using gradle.
-  No sponge or anything else used, only forge essentials client-side mod. Most of these issues have been verified in single player.

I'll try to list more issues as time goes on.

**Comment by Liruxo** [comment]
First thank you for reporting all those bugs. But this leads directly to my problem with this: It would be very kind of you to split this up into a bunch of issue reports, since this issue would very fast become very confusing. Also if you are using "Latest build off github using gradle." as version, please add the date this was the "latest" since this changes very fast.

Concerning the reported bugs: There seem to be a lot of duplicates here, e.g. #1037
Additional advises:
- For reporting crashes, please use the correct formula, because pasting one line of the crash report sadly doesn't help me (or anyone else) very much
- Please be more specific: I personally cant tell what a "Russian self-propelled artillery" is. From which content pack is this? What is the item name?

**Comment by matt-bel** [comment]
The Content Pack is WWII, the vehicle is SU-1-12, this is the line. The date was May 3 2019
```
//Secondary gun
AllowAllAmmo False
AddAmmo ArtilleryHEShell
```

And I'll keep it in mind for the future to report bugs separately. In terms of the crash, again, it wasn't I who caught it, my staff reported it to me. I'll eventually go back to check the logs and find the error when it happened and paste it in full. Using the teams command to set despawn timer to 0 (never) fixed the issue.

**Comment by ChrisLane** [comment]
Fixed the MG42 skin in 2c5a0247dd1ecd628448a252d9df4c8b6b0b56ea

**Comment by ChrisLane** [comment]
Fixed MGs using whole ammo stack in 59f702f

**Comment by ChrisLane** [comment]
Closing this in favour of separate issues.
